build_tools/aws-sdk-code-generator/spec/protocol_tests_spec_helper.rb

```diff
@@ -314,3 +322,11 @@ def error_case?(test_case)
     end
   end
 end
+
+# @api private
+# RSpec::Matchers.define :match_cbor_float do |expected|
```

Comment:
I think we probably want something almost exactly like the param matcher we have in hearth: https://github.com/smithy-lang/smithy-ruby/blob/main/hearth/lib/hearth/query/param_matcher.rb

We can add it to core's spec_helper (either in that file or as a separate file thats included).  And then we can use it on the cbor decode tests as well.  I think it can handle float precision and NaN equality.

Reply:
I don't think a matcher makes sense because it's not reusable across specs. A simple method to iterate data members seems best so I did that.